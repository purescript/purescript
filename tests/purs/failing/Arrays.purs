-- @shouldFailWith TypesDoNotUnify
module Main where

foreign import ix :: forall a. Array a -> Int -> a

test = \arr -> arr `ix` (0 `ix` 0)
