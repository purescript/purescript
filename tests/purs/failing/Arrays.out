Error found:
in module [33mMain[0m
at tests/purs/failing/Arrays.purs:6:26 - 6:27 (line 6, column 26 - line 6, column 27)

  Expected type
  [33m          [0m
  [33m  Array t0[0m
  [33m          [0m
  but found type
  [33m     [0m
  [33m  Int[0m
  [33m     [0m

while checking that expression [33m0[0m
  has type [33mArray t0[0m
while applying a function [33mix[0m
  of type [33mArray t0 -> Int -> t0[0m
  to argument [33m0[0m
in value declaration [33mtest[0m

where [33mt0[0m is an unknown type

See https://github.com/purescript/documentation/blob/master/errors/TypesDoNotUnify.md for more information,
or to contribute content related to this error.

